ircd/s_user.c

```diff
@@ -972,6 +972,33 @@ report_and_set_user_flags(struct Client *source_p, struct ConfItem *aconf)
 	}
 }
 
+void
+report_priv_change(struct Client *client, struct PrivilegeSet *old, struct PrivilegeSet *new)
+{
+	const struct PrivilegeSet *added, *removed, *unchanged;
+	if (old == NULL && new == NULL)
+	{
+		new = client->user->privset;
+		if (new != NULL)
+			old = new->shadow;
+	}
+
+	const struct PrivilegeSet **result = privilegeset_diff(old, new);
```

Comment:
I think these three results should get freed at some point. They are allocated with new_orphan which mallocs

Reply:
I think I'd prefer not to; currently they're allocated once and then reused for the life of the program, so I think of them as morally static objects that happen to be allocated in practice.

Reply:
I missed that there was a test such that these are allocated only once